### PR TITLE
Feature/validate search query

### PR DESCRIPTION
**Pull request by hayleyw7**
# Description drawing

- Added data validation for ID
- Fetched foundPokemon URL
- Assigned foundPokemon's name (aka data.name in the fetch) to App.state.foundPokemon
    - ^^ This was just a test that prints out the value assigned to the name key from the singlePokemon object that the API just fetch. foundPokemon.name WILL need to be displayed, but so will many other keys/values in foundPokemon. It will be an easy adjustment, but hopefully just printing the name to the console will help you see the steps and what's happening.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] local host 3000
- [ ] console.log()
- [ ] dev tools
- [ ] cypress testing

# Checklist:

- [x] My code follows the style guidelines of this project/linted
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -23,26 +25,63 @@ class App extends Component {
...
+    } else {
+      console.log('No Good ID!')
+      return 'No Good ID!'
+    }
+ }
+ 
+  validatePokemonData = (queriedPokemon) => {
+    if (!isNaN(queriedPokemon.queriedPokemon)) {
+      const idValidated = this.validatePokemonID(queriedPokemon.queriedPokemon)
+      this.pokemonApiFetch(idValidated)
```
Comment (Matt-Roden) [inline src/components/App/App.js:81]: I love that you broke this up into more modular functions with single responsibilities.

File: src/components/App/App.js [hunk 2]
```diff
@@ -23,26 +25,63 @@ class App extends Component {
...
+    })
+    console.log(this.state.foundPokemon)
+    return this.state.foundPokemon
+  }
+
+  validatePokemonID(queriedPokemon) {
+    console.log('reaches id function')
+    const parsedID = parseInt(queriedPokemon);
+
+    if (parsedID > 0 && parsedID < 152) {
```
Comment (Matt-Roden) [inline src/components/App/App.js:66]: Gen1 only!

File: src/apiCalls.js [hunk 3]
```diff
@@ -2,3 +2,8 @@ export const getPokedexData = () => {
   return fetch('https://pokeapi.co/api/v2/pokemon?limit=151')
     .then(res => res.ok ? res.json() : this.displayErrorInfo(res))
 }
+
+export const getPokemonData = (suffix) => {
+  return fetch(`https://pokeapi.co/api/v2/pokemon/${suffix}`)
+    .then(res => res.ok ? res.json() : this.displayErrorInfo(res))
+}
```
Comment (Matt-Roden) [inline src/apiCalls.js:9]: nice job starting work on incorporating a fetch to round off this logic. Eventually this should probably get called in the component did mount method for our pokemon details page.

**Review by Matt-Roden — APPROVED** [review]
Great work. We will get this little bugs figured out, but this is super helpful in keeping our project moving! 💪

**Comment by hayleyw7** [comment]
Went with other code

**Pull request closed without merging**